modules/imgcodecs/src/grfmt_spng.cpp

```diff
@@ -526,6 +530,8 @@ bool SPngEncoder::write(const Mat &img, const std::vector<int> &params)
     int width = img.cols, height = img.rows;
     int depth = img.depth(), channels = img.channels();
     volatile bool result = false;
+    bool set_compression_level = false;
```

Comment:
those two should not be outside `if (ctx)` below right? I guess they should be kept right after `ihdr.width = width;` no ?